src/main/java/seedu/address/logic/commands/CommandResult.java

```diff
@@ -38,6 +48,19 @@ public CommandResult(String feedbackToUser) {
         this(feedbackToUser, false, false, false);
     }
 
+    /**
+     * Constructs a {@code CommandResult} with the specified {@code feedbackToUser},
+     * also instructs window to display an internship application at specified {@code internshipApplicationIndex}.
+     */
+    public CommandResult(String feedbackToUser, InternshipApplication internshipApplicationToShow) {
+        this.feedbackToUser = requireNonNull(feedbackToUser);
+        this.showInternshipApplication = true;
+        this.internshipApplicationToShow = requireNonNull(internshipApplicationToShow);
+        showHelp = false;
+        showStatistics = false;
+        exit = false;
```

Comment:
Why does CommandResult need to know the internshipApplication to show?
Why not just set it in the model?

Reply:
So two reasons why I ended up choosing this implementation:

1. The Display handling can only be done in MainWindow and it only has a direct association with Logic. Hence, I think it makes it more streamlined if i were to handle it in MainWindow's execution method.
2. At first I wanted to use Index and acquire the IA from getFilteredList in Logic. However, I realised this isn't feasible with add command. As you can add an internship application that is filtered out by the list, which gives an IndexOutOfBoundsException